Doc/c-api/module.rst

```diff
@@ -479,12 +479,35 @@ state:
    .. versionadded:: 3.10
 
 
+.. c:function:: int PyModule_Add(PyObject *module, const char *name, PyObject *value)
+
+   Add an object to *module* as *name*.  This is a convenience function which
+   can be used from the module's initialization function.
+
+   On success, return ``0``. On error, raise an exception and return ``-1``.
+
+   Return ``-1`` if *value* is ``NULL``. It must be called with an exception
+   raised in this case.
+
+   This function "steals" a reference to *value*.  It can be called with
+   a result of function that returns a new reference without bothering to
+   check its result or even saving it to a variable.
+
+   Example usage::
+
+        if (PyModule_Add(module, "spam", PyBytes_FromString(value)) < 0) {
+            goto error;
+        }
+
+   .. versionadded:: 3.10
+
+
 .. c:function:: int PyModule_AddObject(PyObject *module, const char *name, PyObject *value)
 
-   Similar to :c:func:`PyModule_AddObjectRef`, but steals a reference to
+   Similar to :c:func:`PyModule_Add`, but only steals a reference to
    *value* on success (if it returns ``0``).
 
-   The new :c:func:`PyModule_AddObjectRef` function is recommended, since it is
+   The new :c:func:`PyModule_Add` function is recommended, since it is
```

Comment:
What do you think about renaming `PyModule_Add` to `PyModule_AddNew` and `PyModule_AddObjectRef` to `PyModule_Add`? I prefer having a single function, since former function is more convenient for new references and the latter one can be easily expressed in terms of the former one + `Py_XINCREF()`. But if add two functions, it would be better if they have names formed by the same scheme.

Reply:
I'm not good at naming things. Since PyModule_AddObjectRef() was added to Python 3.10, it's no released yet and we can rename it. If we add PyObject_Add(), I would prefer to rename it. I'm with with:

* PyModule_Add() (call Py_INCREF on success) -- current PyModule_AddObjectRef()
* PyModule_AddNew() (always call Py_DECREF)